Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -778,13 +933,33 @@ extension NIOSSLHandler {
                 let ourPromise: EventLoopPromise<Void>? = promises.flattenPromises(on: context.eventLoop)
                 self.writeDataToNetwork(context: context, promise: ourPromise)
             }
+
+            // We detected a .closeOutput action in our action buffer. This means we
+            // close the output after we have written all pending writes.
+            if invokeCloseOutput {
+                self.state = .outputClosed
+                self.doShutdownStep(context: context)
+                self.discardBufferedActions(reason: ChannelError.outputClosed)
+            }
         } catch {
             // We encountered an error, it's cleanup time. Close ourselves down.
-            channelClose(context: context, reason: error)
+            switch self.state {
+            case .outputClosed:
+                break
```

Comment:
Why do we suppress error handling in this state?

Reply:
Fixed 🔨